Mimir.Worker/Services/ItemProductCalculationService.cs

```diff
@@ -0,0 +1,94 @@
+using Bencodex.Types;
+using Lib9c.Models.Extensions;
+using Lib9c.Models.Items;
+using Lib9c.Models.Market;
+using Nekoyume.Battle;
+using Nekoyume.Helper;
+using Nekoyume.TableData;
+using Nekoyume.TableData.Crystal;
+using ILogger = Serilog.ILogger;
+
+namespace Mimir.Worker.Services;
+
+public class ItemProductCalculationService : IItemProductCalculationService
+{
+    private readonly MongoDbService _store;
+    private readonly ILogger _logger;
+
+    public ItemProductCalculationService(MongoDbService store, ILogger logger)
+    {
+        _store = store;
+        _logger = logger;
+    }
+
+    public async Task<int?> CalculateCombatPointAsync(ItemProduct itemProduct)
+    {
+        try
+        {
+            var costumeStatSheet = await _store.GetSheetAsync<CostumeStatSheet>();
+
+            if (costumeStatSheet != null)
+            {
+                int? cp = itemProduct.TradableItem switch
+                {
+                    ItemUsable itemUsable
+                        => CPHelper.GetCP(
+                            (Nekoyume.Model.Item.ItemUsable)
+                            Nekoyume.Model.Item.ItemFactory.Deserialize(
+                                (Dictionary)itemUsable.Bencoded
+                            )
+                        ),
+                    Costume costume => CPHelper.GetCP(new Nekoyume.Model.Item.Costume((Dictionary)costume.Bencoded), costumeStatSheet),
+                    _ => null
+                };
+                return cp;
+            }
+        }
+        catch (Exception ex)
+        {
+            _logger.Error("Error calculating combat point for itemProduct {ItemProductProductId}: {ExMessage}",
```

Comment:
I moved the try/catch block up, but the nullable return value is still mandatory, because the calculations could be without result. E.g. for Crystals if the item is not an Equipment